internal/api/api.go

```diff
@@ -84,19 +115,46 @@ func (api *idler) IsIdle(w http.ResponseWriter, r *http.Request, ps httprouter.P
 	json.NewEncoder(w).Encode(s)
 }
 
-func (api *idler) Info(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+func (api *idler) ClusterDNSView(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 	w.Header().Set("Content-Type", "application/json")
-	ns := ps.ByName("namespace")
 
-	err := json.NewEncoder(w).Encode(api.controller.GetUser(ns))
+	clusterDNSView := api.clusterView.GetDNSView()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(clusterDNSView)
+}
+
+func (api *idler) Info(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	w.Header().Set("Content-Type", "application/json")
+	namespace := ps.ByName("namespace")
+
+	userIdler, ok := api.userIdlers.Load(namespace)
+	if ok {
+		err := json.NewEncoder(w).Encode(userIdler.GetUser())
+
+		if err != nil {
+			log.Error("Could not serialize users")
+			fmt.Fprint(w, "{'error': 'Could not serialize users'}")
+		}
+	} else {
+		w.WriteHeader(http.StatusNotFound)
+	}
+}
 
-	if err != nil {
-		log.Error("Could not serialize users")
-		fmt.Fprint(w, "{'error': 'Could not serialize users'}")
+func (api *idler) getURLAndToken(r *http.Request) (string, string, error) {
+	var openShiftAPIURL string
+	values, ok := r.URL.Query()[OpenShiftAPIParam]
+	if !ok || len(values) < 1 {
+		// TODO Issue #107
```

Comment:
Let's get rid of all TODO notes once this PR is ready (not WIP). I have seen a many TODO notes in the master as well.

Reply:
Sure, that's why this PR is still in WIP